ompi/mca/osc/ucx/osc_ucx_request.h

```diff
@@ -29,15 +30,19 @@ typedef struct ompi_osc_ucx_internal_request {
 #define OMPI_OSC_UCX_REQUEST_ALLOC(win, req)                            \
     do {                                                                \
         opal_free_list_item_t *item;                                    \
+        OPAL_THREAD_LOCK(&mca_osc_ucx_component.lock);                  \
         do {                                                            \
             item = opal_free_list_get(&mca_osc_ucx_component.requests); \
             if (item == NULL) {                                         \
                 if (mca_osc_ucx_component.ucp_worker != NULL &&         \
                     mca_osc_ucx_component.num_incomplete_req_ops > 0) { \
+                    OPAL_THREAD_UNLOCK(&mca_osc_ucx_component.lock);    \
```

Comment:
Wait, is ucp_worker_progress() not thread safe or is this protecting something else? Kind of want to know because it will affect the btl/ucx design.

Reply:
ucp_worker_progress() is thread safe. Here I unlock before ucp_worker_progress() and lock after it, because I want to protect mca_osc_ucx_component.